client/src/pages/SignUpModal.tsx

```diff
@@ -1,37 +1,121 @@
-import React, { FunctionComponent, useState } from 'react';
+import './Modal.scss';
+import { OverlayTrigger, Popover } from 'react-bootstrap';
 import CommonModal from '../components/organisms/Modal';
+import { createNewAccount } from '../services/auth';
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import {useState} from 'react';
 
-const SignUpModal: FunctionComponent = () => {
+function SignUpModal() {
     const [email, setEmail] = useState("");
     const [password, setPassword] = useState("");
-    const [show, setShow] = useState(true);
-
+    const [hasOneLowerCase, setHasOneLowerCase] = useState(false);
+    const [hasOneUpperCase, setHasOneUpperCase] = useState(false);
+    const [hasOneNumber, setHasOneNumber] = useState(false);
+    const [hasOneSymbol, setHasOneSymbol] = useState(false);
+    const [hasTwelveCharacterMin, setHasTwelveCharacterMin] = useState(false);
     const handleClose = () => setShow(false);
+    const initialReq : string[] = ["at least 1 lowercase letter", "at least 1 uppercase letter","at least 1 number","at least 1 symbol","12 characters minimum" ]
+    const [requirements, setRequirements] = useState(initialReq);
+    const [show, setShow] = useState(true);
+    const [redirectToHome, setRedirectToHome] = useState(false);
+    const requirementsAreFulfilled = !hasOneLowerCase || !hasOneUpperCase || !hasOneNumber || !hasOneSymbol || !hasTwelveCharacterMin;
 
-    const handleClick = async (e: React.FormEvent<HTMLFormElement>) => {
+    const handleClick = async (e:any) => { 
```

Comment:
lint will fail if you use type `any`, speaking from personal experience, so i recommend you keep the `React.FormEvent<HTMLFormElement>` type